File: TFL_example_code/Java/TicTacToeAssign.java
import java.util.Scanner;

public class TicTacToeAssign {
    private char[][] board;
    private char currentPlayer;
    private static final int SIZE = 3;
    private Scanner scanner;

    public TicTacToeAssign() {
        board = new char[SIZE][SIZE];
        currentPlayer = 'X';  // Player 1 starts with 'X'
        scanner = new Scanner(System.in);
        initializeBoard();
    }

    // Initialize the board with empty spaces
    //Each board tile should be assigned en empty char to start like this: ' '
    private void initializeBoard() {

    }

    // Main game loop
    public void play() {
        boolean gameWon = false;
        int moves = 0;

        //While the game is not won and there are still moves to be made 
        //Display the board
        //Make a move
        //Check if the game has been won
        //If the game has not been won switch the player
        //Increment the number of moves


        //Display the board
        //If the game has been won display the winning player
        //If the game was a draw display that it was a draw


    }

    // Display the current state of the board
    private void displayBoard() {
        System.out.println("Current board:");

        for (int i = 0; i < SIZE; i++) {
            for (int j = 0; j < SIZE; j++) {
                System.out.print(board[i][j]);
                if (j < SIZE - 1) System.out.print(" | ");
            }
            System.out.println();
            if (i < SIZE - 1) System.out.println("---------");
        }
    }

    // Prompt the player to make a move
    private void makeMove() {
        boolean validMove = false;

        //While the move is not valid
        while (!validMove) {
            //System.out.println("Player " + currentPlayer + ", enter your move (row and column): ");
            //Get the row and column from the player using scanner
            


            // Check for valid input
            //If the row and column are within the bounds of the board and the tile is empty
            //Set the tile to the current player
            //Set validMove to true
            //Otherwise print that the move is not valid

        }
    }

    // Check if the current player has won the game
    private boolean checkWin() {
        // Check rows, columns, and diagonals for a win
    }

    private boolean checkRows() {
        //Check each row for a win

    }

    private boolean checkColumns() {
        //Check each column for a win
    }

    private boolean checkDiagonals() {
        //Check the diagonals for a win
    }
}

File: TFL_example_code/python/tic_tac_toe_assign.py
# tic_tac_toe.py

from typing import List, Optional, Tuple

def display_board(board: List[str]) -> None:
    """Prints the current board to the terminal."""
    for i in range(3):
        print(" | ".join(board[i*3:(i+1)*3]))
        if i < 2:
            print("-" * 9)
    
def is_valid_move(board: List[str], row: int, col: int) -> bool:
    """Checks if a move is valid (index is within range and the cell is empty)."""
    pass

def make_move(board: List[str], row: int, col: int, player: str) -> None:
    """Applies the player's move to the board."""
    pass

def get_winner(board: List[str]) -> Optional[str]:
    """Returns 'X' or 'O' if there is a winner, or None if there is no winner yet."""
    pass

def is_full(board: List[str]) -> bool:
    """Returns True if the board is full."""
    pass

def get_player_input(player: str, board: List[str]) -> Tuple[int, int]:
    """Prompts the current player for a valid move as row and column."""
    pass

def play_game() -> None:
    """Runs the main game loop."""
    board = [" "] * 9
    current_player = "X"
    winner = None

    while not is_full(board) and not winner:
        display_board(board)
        row, col = get_player_input(current_player, board)
        make_move(board, row, col, current_player)
        winner = get_winner(board)
        current_player = "O" if current_player == "X" else "X"

    display_board(board)
    if winner:
        print(f"Player {winner} wins!")
    else:
        print("It's a tie!")

if __name__ == "__main__":
    play_game()
